frame/core-fellowship/src/lib.rs

```diff
@@ -0,0 +1,593 @@
+		///
+		/// - `origin`: A `Signed` origin of an inducted and ranked account.
+		/// - `wish`: The stated desire of the member.
+		/// - `evidence`: A dump of evidence to be considered. This should generally be either a
+		///   Markdown-encoded document or a series of 32-byte hashes which can be found on a
+		///   decentralised content-based-indexing system such as IPFS.
+		#[pallet::weight(T::WeightInfo::submit_evidence())]
+		#[pallet::call_index(7)]
+		pub fn submit_evidence(
+			origin: OriginFor<T>,
+			wish: Wish,
+			evidence: Evidence<T, I>,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+			ensure!(Member::<T, I>::contains_key(&who), Error::<T, I>::NotTracked);
+			let replaced = MemberEvidence::<T, I>::contains_key(&who);
+			MemberEvidence::<T, I>::insert(&who, (wish, evidence));
+			Self::deposit_event(Event::<T, I>::Requested { who, wish });
+			Ok(if replaced { Pays::Yes } else { Pays::No }.into())
+		}
+
+		/// Introduce an already-ranked individual of the collective into this pallet. The rank may
+		/// still be zero.
+		///
+		/// This resets `last_proof` to the current block and `last_promotion` will be set to zero,
+		/// thereby delaying any automatic demotion but allowing immediate promotion.
+		///
+		/// - `origin`: A signed origin of a ranked, but not tracked, account.
+		#[pallet::weight(T::WeightInfo::import())]
+		#[pallet::call_index(8)]
+		pub fn import(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+			ensure!(!Member::<T, I>::contains_key(&who), Error::<T, I>::AlreadyInducted);
+			let rank = T::Members::rank_of(&who).ok_or(Error::<T, I>::Unranked)?;
+
+			let now = frame_system::Pallet::<T>::block_number();
+			Member::<T, I>::insert(
+				&who,
+				MemberStatus { is_active: true, last_promotion: 0u32.into(), last_proof: now },
+			);
+			Self::deposit_event(Event::<T, I>::Imported { who, rank });
+
+			Ok(Pays::No.into())
+		}
+	}
+
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		/// Convert a rank into a `0..RANK_COUNT` index suitable for the arrays in Params.
+		///
+		/// Rank 1 becomes index 0, rank `RANK_COUNT` becomes index `RANK_COUNT - 1`. Any rank not
+		/// in the range `1..=RANK_COUNT` is `None`.
+		pub(crate) fn rank_to_index(rank: RankOf<T, I>) -> Option<usize> {
+			match TryInto::<usize>::try_into(rank) {
+				Ok(r) if r <= RANK_COUNT && r > 0 => Some(r - 1),
+				_ => return None,
+			}
+		}
+
+		fn dispose_evidence(who: T::AccountId, old_rank: u16, new_rank: Option<u16>) {
+			if let Some((wish, evidence)) = MemberEvidence::<T, I>::take(&who) {
+				let e = Event::<T, I>::EvidenceJudged { who, wish, evidence, old_rank, new_rank };
+				Self::deposit_event(e);
+			}
+		}
+	}
+
+	impl<T: Config<I>, I: 'static> GetSalary<RankOf<T, I>, T::AccountId, T::Balance> for Pallet<T, I> {
+		fn get_salary(rank: RankOf<T, I>, who: &T::AccountId) -> T::Balance {
```

Comment:
Nit: I still think the the `rank` on the `get_salary` trait is superfluous and can still be implemented by the implementer in a ranked manner by doing the `Member->Rank` lookup themselve.

Reply:
Thing is that the caller will generally need to ascertain that there *is* a rank prior to calling, so we may as well pass it in.

Reply:
>There are plenty of ideas for implementation which would not necessarily have that lookup available to them.

Okay I see :+1: